### DISCUSSION TITLE
about : coordinate projection

### DISCUSSION THREAD

**Discussion by prince-Ha** [fatiando/community#88] [Q&A] [answered]
Hi, I have a quetion about projection
below is my code

I projected the lon/lat coordinate to mercator 
It works very well 

--------------------------------------------------------------------------------------
projection = pyproj.Proj(proj = 'merc', lat_ts = data.Lat.mean())
easting, northing = projection(data.Lon.values, data.Lat.values)
data = data.assign(easting_m=easting, northing_m = northing)
topography_proj = vd.project_grid(topography, projection, method="nearest")
-----------------------------------------------------------------------------------------------
but when I converted the coordinate lon to utm then, All values became nan.

projection = pyproj.Proj("+proj=utm +zone=52 +ellps=WGS84 +datum=WGS84 +units=m +no_defs ")
easting, northing = projection(data.Lon, data.Lat)
data = data.assign(easting_m=easting, northing_m = northing)
topography_proj = vd.project_grid(topography, projection, method="nearest")
---------------------------------------------

<img width="794" alt="image" src="https://user-images.githubusercontent.com/69688086/205504408-ebac19f2-12f5-4392-a549-7801bd73df4e.png">

---------------------------

Is verde converted from lon to mercator only?

thanks have a nice day

**Comment by santisoler** [comment, marked as answer]
Hi @prince-Ha!

As you can see in the screenshot you posted not every point in the UTM projected grid is nan. Having nans in projected grid is expected: the `vd.project_grid` is basically doing two things:
1. Projects the coordinates of each one of the grid points.
2. Defines a set of regularly spaced projected `easting` and `northing` coordinates and interpolates the projected values of the grid into this new regular grid.

Since projections "deform" the geometry, when projecting a regular grid in longitude and latitude we don't get a regular grid in the projected coordinates. That's why we apply the second step. During this interpolation, some points of the projected grid will be set to nans, particularly the ones located on the boundaries of the grid that are outside the convex-hull region of the grid points. This is ok.

Try to plot the two grids and see how each projection make the projected grid looks like.

Depending on which projection you use, it might generate more or less nans. This is related to how the chosen projection "deforms" the original grid. Nevertheless, the choice of the projection should not be based on the values of nans you get while projecting the grid, but on which is the projection you need to use for the ultimate purposes of your study.

**Reply by prince-Ha** [reply]
I'm so busy that I'm finally reading your reply Thank you for your kind reply always :)
